src/lib/datepicker/README.md

```diff
@@ -1,3 +1,181 @@
+`md-datepicker`. It also has an `opened` property that reflects the status of the popup.
+
+```ts
+@Component({...})
+export class MyComponent implements AfterViewInit {
+  @ViewChild(MdDatepicker) dp: MdDatepicker<Date>;
+  
+  ngAfterViewInit() {
+    dp.open();
+  }
+}
+```
+
+As seen in the example above, the `MdDatepicker` type is generic. This is because it can work with
+different date implementations depending on the `DateAdapter` provided in your app. The generic type
+on the `MdDatepicker` should be the same as the generic type of the `DateAdapter` for your app
+(see section on
+[choosing a date implementation](#choosing-a-date-implementation-and-date-format-settings)).
+
+### Choosing a date implementation and date format settings
+The datepicker was built to be date implementation agnostic. This means that it can be made to work
+with a variety of different date implementations. However it also means that developers need to make
+sure to provide the appropriate pieces for the datepicker to work with their chosen implementation.
+The easiest way to ensure this is just to import one of the pre-made modules (currently
+`MdNativeDateModule` is the only implementation that ships with material, but there are plans to add
+modules for Moment.js and Closure Library support):
+ * `MdNativeDateModule` - support for native JavaScript Date object
+ 
+These modules include providers for `DateAdapter` and `MD_DATE_FORMATS`
+
+```ts
+@NgModule({
+  imports: [MdDatepickerModule, MdNativeDateModule],
+})
+export class MyApp {}
+```
+
+#### Customizing the date implementation
+The datepicker does all of its interaction with date objects via the `DateAdapter`. Making the
+datepicker work with a different date implementation is as easy as extending `DateAdapter`, and
+using your subclass as the provider. You will also want to make sure that the `MD_DATE_FORMATS`
+provided in your app are formats that can be understood by your date implementation.
+
+```ts
+@NgModule({
+  imports: [MdDatepickerModule],
+  providers: [
+    {provide: DateAdapter, useClass: MyDateAdapter},
+    {provide: MD_DATE_FORMATS, useValue: MY_DATE_FORMATS},
+  ],
+})
+export class MyApp {}
+```
+
+#### Customizing the parse and display formats
+The `MD_DATE_FORMATS` object is just a collection of formats that the datepicker uses when parsing
+and displaying dates. These formats are passed through to the `DateAdapter` so you will want to make
+sure that the format objects you're using are compatible with the `DateAdapter` used in your app.
+This example shows how to use the native `Date` implementation from material, but with custom
+formats.
+
+```ts
+@NgModule({
+  imports: [MdDatepickerModule],
+  providers: [
+    {provide: DateAdapter, useClass: NativeDateAdapter},
+    {provide: MD_DATE_FORMATS, useValue: MY_NATIVE_DATE_FORMATS},
+  ],
+})
+export class MyApp {}
+```
+
+### Internationalization
```

Comment:
The reason I didn't do this is that it seems to make more sense to talk about formats when I talk about choosing a date implementation, since they're closely related